internal/provisioner/objects/rbac/rbac.go

```diff
@@ -53,19 +54,44 @@ func ensureContourRBAC(ctx context.Context, cli client.Client, contour *model.Co
 		return fmt.Errorf("failed to ensure service account %s/%s: %w", contour.Namespace, names.ServiceAccount, err)
 	}
 
-	// Ensure cluster role & binding.
-	if err := clusterrole.EnsureClusterRole(ctx, cli, names.ClusterRole, contour); err != nil {
-		return fmt.Errorf("failed to ensure cluster role %s: %w", names.ClusterRole, err)
-	}
-	if err := clusterrolebinding.EnsureClusterRoleBinding(ctx, cli, names.ClusterRoleBinding, names.ClusterRole, names.ServiceAccount, contour); err != nil {
-		return fmt.Errorf("failed to ensure cluster role binding %s: %w", names.ClusterRoleBinding, err)
+	// By default, Contour watches all namespaces, use default cluster role and rolebinding
+	clusterRoleForClusterScopedResourcesOnly := true
+	if contour.WatchAllNamespaces() {
+		// Ensure cluster role & binding.
+		if err := clusterrole.EnsureClusterRole(ctx, cli, names.ClusterRole, contour, !clusterRoleForClusterScopedResourcesOnly); err != nil {
+			return fmt.Errorf("failed to ensure cluster role %s: %w", names.ClusterRole, err)
+		}
+		if err := clusterrolebinding.EnsureClusterRoleBinding(ctx, cli, names.ClusterRoleBinding, names.ClusterRole, names.ServiceAccount, contour); err != nil {
+			return fmt.Errorf("failed to ensure cluster role binding %s: %w", names.ClusterRoleBinding, err)
+		}
+	} else {
+		// Ensure cluster role & cluster binding for gatewayclass and other cluster scope resource first since it's cluster scope variables
+		if err := clusterrole.EnsureClusterRole(ctx, cli, names.ClusterRole, contour, clusterRoleForClusterScopedResourcesOnly); err != nil {
+			return fmt.Errorf("failed to ensure cluster role %s: %w", names.ClusterRole, err)
+		}
+		if err := clusterrolebinding.EnsureClusterRoleBinding(ctx, cli, names.ClusterRoleBinding, names.ClusterRole, names.ServiceAccount, contour); err != nil {
+			return fmt.Errorf("failed to ensure cluster role binding %s: %w", names.ClusterRoleBinding, err)
+		}
+
+		// includes contour's namespace if it's not inside watchNamespaces
+		ns := contour.Spec.WatchNamespaces
+		if !slice.ContainsString(ns, contour.Namespace) {
+			ns = append(ns, contour.Namespace)
+		}
```

Comment:
I think we might want to circle back to this and only give the minimum set of permissions needed in the Gateway/Contour instance's namespace if it's not explicitly included in `WatchNamespaces`, but IMO it's going to be most common for folks to include the Gateway's namespace in `WatchNamespaces` anyway so probably mostly a moot point. I can file a follow-up issue for this.

Reply:
Cool, let me remove that in following PR then. Better let it error out if user missed the Contour's namespace